plugins/pixart-tp/fu-pxi-tp-register.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2025 Harris Tai <harris_tai@pixart.com>
+ * Copyright 2025 Micky Hsieh <micky_hsieh@pixart.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+ #include "config.h"
+
+#include "fu-pxi-tp-register.h"
+
+#define REPORT_ID_SINGLE 0x42
+#define REPORT_ID_BURST	 0x41
+#define REPORT_ID_USER	 0x43
+#define OP_READ		 0x10
+
+// --- System Register ---
+
+gboolean
+fu_pxi_tp_register_write(FuPxiTpDevice *self, guint8 bank, guint8 addr, guint8 val, GError **error)
+{
+	guint8 buf[4] = {REPORT_ID_SINGLE, addr, bank, val};
+	return fu_pxi_tp_common_send_feature(self, buf, sizeof(buf), error);
+}
+gboolean
```

Comment:
I think running ./contrib/reformat-code.py on this subdirectory might be a good idea